### PR TITLE
Wire up x-ui rendering hints in ActionConfigParameterFields (Chunk 3)

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary

- Update `ActionConfigParameterFields` to consume `x-ui` schema extensions: field ordering (`x-ui.order`), custom labels (`x-ui.label`), placeholders (`x-ui.placeholder`), grouped collapsible sections (`x-ui.groups`), conditional visibility (`visible_when`), and widget type rendering via `ParameterFieldWidget`
- Add `placeholder` override prop to `ParameterFieldWidget` so constraint modes (wildcard/pattern) can override `x-ui.placeholder`
- Add comprehensive test suite (26 tests) covering all x-ui features and backwards compatibility

Part of #551

### Claude Code
- [x] All 26 new ActionConfigParameterFields tests pass
- [x] All 31 existing ParameterFieldWidget tests pass
- [x] Full frontend test suite passes (841 tests across 94 files)
- [x] Frontend TypeScript build passes clean
- [x] Verify no regressions in existing connector action config dialogs

### OpenClaw
- [ ] Review grouped/ordered rendering UX in the browser
- [ ] Verify collapsible group sections look correct with Stripe/Trello connector x-ui hints
- [ ] Confirm backwards compatibility with connectors that don't use x-ui

https://claude.ai/code

### CODE REVIEW

File: frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx [hunk 1]
```diff
@@ -45,100 +54,226 @@ export function ActionConfigParameterFields({
...
+  const groups = parametersSchema["x-ui"]?.groups;
 
   function getMode(key: string): ParamMode {
     return modes[key] ?? inferModeFromValue(values[key] ?? "");
   }
 
+  function renderField(key: string) {
+    const prop = properties[key];
+    if (!prop) return null;
+    if (!isFieldVisible(prop, values)) return null;
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx:67]: Fixed in 5ce09ab. Updated `isFieldVisible()` in `parameterSchema.ts` to coerce string form values when comparing against boolean/number `equals` rules: `String(fieldValue) === String(rule.equals)` for booleans, `Number(fieldValue) === rule.equals` for numbers. Added 3 tests covering boolean and number coercion.

File: frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx [hunk 2]
```diff
@@ -45,100 +54,226 @@ export function ActionConfigParameterFields({
...
+        {ungroupedKeys.map(renderField)}
+        {groups.map((group) => {
+          const groupKeys = keysByGroup.get(group.id) ?? [];
+          if (groupKeys.length === 0) return null;
+          return (
+            <CollapsibleFieldGroup key={group.id} group={group}>
+              {groupKeys.map(renderField)}
+            </CollapsibleFieldGroup>
+          );
+        })}
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/ActionConfigParameterFields.tsx:137]: Fixed in 5ce09ab. Added a `visibleGroupKeys` filter that checks `isFieldVisible()` for each group's fields before deciding whether to render the group. Empty groups (all fields hidden by `visible_when`) are now skipped entirely. Added a test for this edge case.

File: frontend/src/pages/agents/connectors/__tests__/ActionConfigParameterFields.test.tsx [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+    it("collapses an expanded group when clicked", async () => {
+      const user = userEvent.setup();
+      renderFields(groupedSchema);
+
+      // Billing starts expanded
+      expect(screen.getByText("Customer")).toBeInTheDocument();
+
+      // Click to collapse
+      await user.click(screen.getByText("Billing"));
+
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/pages/agents/connectors/__tests__/ActionConfigParameterFields.test.tsx:316]: Good suggestion. The existing tests did pass since RTL can match text nodes, but using `getByRole("button", { name: /Options/ })` is more robust. Updated all group button queries in 92d2aca.

**Comment by chiedoclaude** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### PR Checklist Items
- ✅ **`Verify no regressions in existing connector action config dialogs`** — Ran all 6 connector test suites (105 tests) and all 10 dashboard test suites (95 tests) — all pass with no regressions (`n/a (verification only)`)

### Decision Log

#### ✅ Implemented
- greptile-apps[bot] asked for Fix visible_when boolean/number equals values that never match form string values → implemented in `5ce09ab`
- greptile-apps[bot] asked for Fix empty group rendering when all fields are conditionally hidden by visible_when → implemented in `5ce09ab`
- greptile-apps[bot] asked for Use getByRole queries for group button assertions instead of getByText for robustness → implemented in `92d2aca`

---
*Watch session ended after 6 minutes of inactivity. Processed 3 comments across 6 poll cycles.*

**Comment by chiedoclaude** [comment]
⚠️ **Auto-merge failed.** The `--automerge` flag was set, but the merge could not be completed. Please merge manually.
